DNS/DNS_Cache.c

```diff
@@ -0,0 +1,297 @@
+                                               &ulIPAddress,
+                                               0,
+                                               pdTRUE );
+        #endif
+        return ulIPAddress;
+    }
+
+    BaseType_t FreeRTOS_dns_update( const char * pcName,
+                                    uint32_t * pulIP,
+                                    uint32_t ulTTL )
+    {
+        ( void ) FreeRTOS_ProcessDNSCache( pcName,
+                                           pulIP,
+                                           0,
+                                           pdFALSE );
+    }
+
+    void FreeRTOS_dnsclear( void )
+    {
+        ( void ) memset( xDNSCache, 0x0, sizeof( xDNSCache ) );
+        xFreeEntry = 0;
+    }
+
+/**
+ * @brief Send a DNS message to be used in NBNS or LLMNR
+ *
+ * @param[in] pcName: the name of the host
+ * @param[in,out] pulIP: when doing a lookup, will be set, when doing an update,
+ *                       will be read.
+ * @param[in] ulTTL: Time To Live
+ * @param[in] xLookUp: pdTRUE if a look-up is expected, pdFALSE, when the DNS cache must
+ *                     be updated.
+ *
+ * @return
+ */
+    BaseType_t FreeRTOS_ProcessDNSCache( const char * pcName,
+                                         uint32_t * pulIP,
+                                         uint32_t ulTTL,
+                                         BaseType_t xLookUp )
+    {
+        BaseType_t x;
+        BaseType_t xFound = pdFALSE;
+        TickType_t xCurrentTickCount = xTaskGetTickCount();
+        uint32_t ulCurrentTimeSeconds;
+        uint32_t ulIPAddressIndex = 0;
+
+        configASSERT( ( pcName != NULL ) );
+
+        ulCurrentTimeSeconds = ( xCurrentTickCount / portTICK_PERIOD_MS ) / 1000UL;
+        x = prvFindEntryIndex( pcName );
+
+        if( x != -1 )
+        { /* Element found */
+            if( xLookUp == pdTRUE )
+            {
+                prvGetCacheIPEntry( x,
+                                    pulIP,
+                                    ulCurrentTimeSeconds );
+            }
+            else
+            {
+                prvUpdateCacheEntry( x,
+                                     ulTTL,
+                                     pulIP,
+                                     ulCurrentTimeSeconds );
+            }
+        }
+        else /* Element not Found */
+        {
+            if( xLookUp == pdTRUE )
+            {
+                *pulIP = 0UL;
+            }
+            else
+            {
+                prvInsertCacheEntry( pcName,
+                                     ulTTL,
+                                     pulIP,
+                                     ulCurrentTimeSeconds );
+            }
+        }
+
+        if( ( xLookUp == pdFALSE ) || ( *pulIP != 0UL ) )
+        {
+            FreeRTOS_debug_printf( ( "prvProcessDNSCache: %s: '%s' @ %lxip\n", ( xLookUp != 0 ) ? "look-up" : "add", pcName, FreeRTOS_ntohl( *pulIP ) ) );
+        }
+
+        return !!( x + 1 );
```

Comment:
I understand that this is a true 'C' style code. But can we use an if here? It will make it clear to the reader in my opinion.
```
if( x == -1 )
{
    xReturn = 0;
}
else
{
    xReturn = 1;
}
```